src/ensembl/src/content/app/species/state/sidebar/speciesSidebarSlice.ts

```diff
@@ -94,6 +99,26 @@ export const fetchSidebarPayload = (): ThunkAction<
   );
 };
 
+export const toggleSidebarAndSave = (): ThunkAction<
+  void,
+  any,
+  null,
+  Action<string>
+> => (dispatch, getState: () => RootState) => {
+  const state = getState();
+  const activeGenomeId = getActiveGenomeId(state);
+  if (!activeGenomeId) {
+    return;
+  }
+  const isSidebarCurrentlyOpen = isSidebarOpen(state);
+
+  speciesStorageService.updateSidebarState(
+    set(`isOpen`, !isSidebarCurrentlyOpen, state.speciesPage.sidebar)
+  );
```

Comment:
I forget: do we need the whole `SpeciesPageSidebarState` here? Won't the `Partial<SpeciesPageSidebarState>` (just `{ isOpen }`) do?

Reply:
Yes, `Partial<SpeciesPageSidebarState>` will do. I have updated it.

Reply:
But now instead of the whole line 115 (`set(`isOpen`, !isSidebarCurrentlyOpen, state.speciesPage.sidebar)`) you can just write `speciesStorageService.updateSidebarState({ isOpen: !isSidebarCurrentlyOpen })`

Reply:
That's done